Softeq.XToolkit.WhiteLabel/Validation/ValidatableObject.cs

```diff
@@ -0,0 +1,80 @@
+// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using Softeq.XToolkit.Common;
+using Softeq.XToolkit.Common.Collections;
+
+namespace Softeq.XToolkit.WhiteLabel.Validation
+{
+    public class ValidatableObject<T> : ObservableObject, IValidatableObject
+    {
+        private readonly IList<IValidationRule<T>> _validationRules;
+
+        private T _value = default!;
+        private bool _isValid = true;
+
+        public ValidatableObject()
+        {
+            _validationRules = new List<IValidationRule<T>>();
+
+            Errors = new ObservableRangeCollection<string>();
+            Errors.CollectionChanged += ErrorsCollectionChanged;
+
+            CleanErrorsOnChange = true;
+        }
+
+        public bool CleanErrorsOnChange { get; set; }
+
+        public T Value
+        {
+            get => _value;
+            set
+            {
+                if (Set(ref _value, value) && CleanErrorsOnChange)
+                {
+                    Errors.Clear();
+                }
+            }
+        }
+
+        public bool IsValid
+        {
+            get => _isValid;
+            private set => Set(ref _isValid, value);
+        }
+
+        public ObservableRangeCollection<string> Errors { get; }
```

Comment:
Is it intended that Errors is may be modifiable from outside? Seems like it should be controlled by internal logic only

Reply:
ValidableObject control only internal validators, errors collection is a validators output, and can be changed externally after for example any server validators response and etc.)

Reply:
For this case, maybe it is better to add some kind of `AddError(string)` method to `ValidatableObject`?

Reply:
I added `AddError(string)` previously and revert changes, because in this case, we have several ways to add errors. Also, we have cases when needs reset errors from external.

Reply:
We may add `Clear()` method for that) My idea is that class should explicitly describe it capabilities through interface. In my opinion, it is not really obvious that you need to manipulate the collection in order to achieve certain behavior. Without looking into source code, at least.
Another idea, which is definitely more complicated, but probably more correct in terms of abstraction model - is to add support for async validation rules.

Reply:
I agree to add `ClearErrors()` and `AddError()` but in this case, we still will have two approaches to use errors.

Async rules approach not so often used in previous projects. Also, need another interface and implementation, can be overhead for now.
@TimofeyBurak Maybe do you have an example of async rules?

Reply:
@TimofeyBurak I think we already have a public interface for manage ValidatableObject -> Errors -> Add / Clear
case ValidatableObject -> AddError would be redundant for object